### DISCUSSION TITLE
Expose data quality & recency status to BI tools

### DISCUSSION THREAD

**Discussion by geoHeil** [dagster-io/dagster#18061] [Ideas]
We want to render a green checkmark in Tableau dashboards to show that data quality is fine and data is fresh. In DBT cloud some HTML snippet is rendered which could be included. As Dagster is already aware of these data points via asset checks and regular materializations - is there some way to expose this to tableau?

**Comment by alangenfeld** [comment]
The information is available over the GraphQL API, can that be used in the context of tableau?

**Reply by geoHeil** [reply]
I asked a friend / tableau expert. His response roughly was:

It should be possible in concept with Tableau’s Table Extensions. See https://help.tableau.com/current/pro/desktop/en-us/td_table_extensions.htm for more information on Table Extensions.

In the worst case, you might have to use the Tableau Extracts API (or something similarly named) to write Tableau's own Hyper File and then integrate it into the dashboard - probably as its own small sheet.

The API can also be used with Python, as far as I remember. 

… or create a Web Data Connector with the corresponding API and use it in Tableau.

… these options have probably been greatly simplified through the Table Extensions.

… but you can program quite a bit (if you want/need to), see the API list under Help > Developer Resources: https://www.tableau.com/support/help?_gl=1*1twt6sb*_ga*NzgyNzQwMzQ1LjE3MDAyNTM3NzA.*_ga_8YLN0SNXVS*MTcwMDI1Mzc2OS4xLjEuMTcwMDI1NDU2OC4wLjAuMA..&_ga=2.250121028.374410386.1700253770-782740345.1700253770 for more details.

… but directly, you won’t be happy with JSON, XML, etc., as data formats in Tableau, because at the end of the day, Tableau wants tables and relationships.


TLDR this means for me there is no simple happy path.

DBT cloud provides a HTML snippet with direct easy integration.

I think it would be a beneficial feature of Dagster cloud if it could provide such https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge status badge features for data assets that change color based on rules (freshness SLA and quality checks)
